### PR TITLE
[CN-643] Allow to create HotBackup before Hazelcast cluster is ready

### PR DESCRIPTION
**Pull request by dzeromski-hazelcast**
## Description

Do not return error when Hazelcast CR is not ready yet.

### CODE REVIEW

File: test/e2e/hazelcast_persistence_test.go [hunk 1]
```diff
@@ -252,4 +252,25 @@ var _ = Describe("Hazelcast CR with Persistence feature enabled", Label("hz_pers
...
+		setLabelAndCRName("hp-8")
+
+		clusterSize := int32(1)
+		hazelcast := hazelcastconfig.HazelcastPersistencePVC(hzLookupKey, clusterSize, labels)
+		hazelcast.Spec.Persistence.ClusterDataRecoveryPolicy = hazelcastv1alpha1.MostRecent
+
+		By("creating HotBackup CR")
+		hotBackup := hazelcastconfig.HotBackup(hbLookupKey, hazelcast.Name, labels)
+		Expect(k8sClient.Create(context.Background(), hotBackup)).Should(Succeed())
+
```
Comment (SeriyBg) [inline test/e2e/hazelcast_persistence_test.go:269]: I would add an assert here for the `HotBackupNotStarted` status

**Review by SeriyBg — APPROVED** [review]
One small suggestion, but LGTM

File: test/e2e/hazelcast_persistence_test.go (hunk 1 above)
Comment (dzeromski-hazelcast) [reply test/e2e/hazelcast_persistence_test.go:269]: I was thinking about it and decided no to do it. Do we care about this status? We just want backup to start and dont care what is the status before.

Comment (SeriyBg) [reply test/e2e/hazelcast_persistence_test.go:269]: Well, IMO we do care that the `HotBackup` is started even after being set to the `NotStarted` state. So this way will eliminate the rare but possible case when the HB reconcile is executed after the `Hazelcast CR` is created

Comment (dzeromski-hazelcast) [reply test/e2e/hazelcast_persistence_test.go:269]: Is it a bug? Running HB reconcile is executed after the Hazelcast CR is created shouldn't be a problem, right?

Comment (SeriyBg) [reply test/e2e/hazelcast_persistence_test.go:269]: It shouldn't be a problem, of course. It's just that I would prefer to have an assertion that proves that with this scenario our logic still works fine 😃

**Review by cagric0 — APPROVED** [review]

File: test/e2e/hazelcast_persistence_test.go (hunk 1 above)
Comment (dzeromski-hazelcast) [reply test/e2e/hazelcast_persistence_test.go:270]: Fixed.

**Pull request merged**